### PR TITLE
systemd: introduce a new attribute for services using PrivateDevices=yes

### PR DESCRIPTION
**Pull request by fishilico**
When a service is configured with `PrivateDevices=yes`, its `/dev` directory has label `tmpfs_t`. This requires allowing more accesses in order for the service to use `/dev`.

This is related to https://github.com/SELinuxProject/refpolicy/pull/61

### CODE REVIEW

**Comment by fishilico** [comment]
I changed the name of the interface and removed the attribute.

**Comment by fishilico** [comment]
My patch currently breaks building with `SYSTEMD=n`, as
```
optional_policy(`
	systemd_PrivateDevices(entropyd_t)
')
```

is reduced to:
```
optional {
}
```

The issue here is that `systemd_PrivateDevices` does not require anything but a compile-time boolean to be set. In order for this to work, dropping the `optional_policy` is possible (calling `systemd_PrivateDevices` does not make a module depends on `systemd` module).

How would you like this to be fixed?

**Comment by pebenito** [comment]
Simple fix. Remove the ifdef. That check should be on caller side.

**Comment by pebenito** [comment]
To clarify, since this is the systemd module, all the interfaces should be assuming init_systemd=y. If that check is necessary it should be done on caller side.

**Comment by fishilico** [comment]
Thanks, I moved the ifdef to the caller side.

File: policy/modules/services/entropyd.te [hunk 1]
```diff
@@ -83,3 +83,9 @@ optional_policy(`
 optional_policy(`
 	udev_read_db(entropyd_t)
 ')
+
+ifdef(`init_systemd', `
```
Comment (pebenito) [inline policy/modules/services/entropyd.te]: This should move to line 66.

Comment (fishilico) [reply policy/modules/services/entropyd.te]: Moved, thanks.

**Pull request merged**